pkg/util/metric/stats/counter.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2023 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stats
+
+import "sync/atomic"
+
+// Counter represents a combination of global & local counter.
+type Counter struct {
+	local  atomic.Int64
+	global atomic.Int64
+}
+
+// Add adds to local counter
+func (c *Counter) Add(delta int64) (new int64) {
+	return c.local.Add(delta)
+}
+
+// Load return the sum of local and global
+func (c *Counter) Load() int64 {
+	return c.global.Load() + c.local.Load()
```

Comment:
you can then load local and/or global

Reply:
Right now, the default `Load()` exposes the global count only. In our case, `global count = global + local`.

`LoadC()` exposes the local counter value alone (it also merge and resets the value)

Reply:
Right now, only exposing current window counter, using SwapW().